line-sdk/src/main/java/com/linecorp/linesdk/auth/internal/BrowserAuthenticationApi.java

```diff
@@ -93,14 +94,14 @@ boolean isLineAppAuthentication() {
     Request getRequest(
             @NonNull Context context,
             @NonNull LineAuthenticationConfig config,
-            @NonNull OneTimePassword oneTimePassword,
+            @NonNull PKCECode pkceCode,
             @NonNull LineAuthenticationParams params)
             throws ActivityNotFoundException {
 
         // "state" may be guessed easily but there is no problem as the follows.
-        // In case of LINE SDK, the correctness of "redirect_uri" will be checked with using "otpId"
+        // In case of LINE SDK, the correctness of "redirect_uri" will be checked with using PKCE
         // instead of "state".
-        final String oAuthState = createRandomString(8);
+        final String oAuthState = createRandomAlphaNumeric(16);
```

Comment:
please make `16` a const~~

Reply:
extracted the following constants:

- `BrowserAuthenticationApi.java` :
```
LENGTH_OAUTH_STATE = 16
LENGTH_OPENID_NONCE = 16
```

- `PKCECode.java` :
```
LENGTH_VERIFIER = 64
```

Reply:
@tsairene :
I have fixed it, please review again if you get time.
🙇‍♂️